### PR TITLE
Add names to several ComponentMode components

### PR DESCRIPTION
**Pull request by amzn-ahmadkrm**
Signed-off-by: amzn-ahmadkrm <ahmadkrm@amazon.com>

## What does this PR do?

Adds the component name to the header of component mode:
image      ->   image

This is implemented in the same way as in Gems\LmbrCentral\Code\Source\Shape\EditorSplineComponentMode.cpp

### CODE REVIEW

File: Gems/WhiteBox/Code/Source/EditorWhiteBoxComponentMode.cpp [hunk 1]
```diff
@@ -284,6 +284,11 @@ namespace WhiteBox
         return interactionHandled;
     }
 
+    AZStd::string EditorWhiteBoxComponentMode::GetComponentModeName() const
+    {
+        return "WhiteBox Edit Mode";
```
Comment (hultonha) [inline Gems/WhiteBox/Code/Source/EditorWhiteBoxComponentMode.cpp:289]: nit: Please can we rename this to 'White Box Edit Mode' please, thanks!

**Review by hultonha — APPROVED** [review]
Looks great to me, just need to update the wording of one ever so slightly. Thanks for fixing this! 🙂

**Review by moraaar — APPROVED** [review]

**Review by hultonha — APPROVED** [review]
LGTM :+1:

File: Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BoxComponentMode.cpp [hunk 2]
```diff
@@ -28,4 +28,9 @@ namespace AzToolsFramework
     {
         m_boxEdit.UpdateManipulators();
     }
+
+    AZStd::string BoxComponentMode::GetComponentModeName() const
+    {
+        return "Tube Shape Edit Mode";
+    }
```
[the remark this answers is not in the document]
Comment (amzn-ahmadkrm) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BoxComponentMode.cpp:35]: Good catch, thank you!

**Review by AMZN-daimini — APPROVED** [review]
I am progressively getting less and less convinced of the usefulness of the blue border given the introduction of the switcher... but I'll bring that up in UX discussions. This looks good for now tho :D

File: Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentMode.cpp [hunk 3]
```diff
@@ -319,6 +319,11 @@ namespace PhysX
         return ids;
     }
 
+    AZStd::string JointsComponentMode::GetComponentModeName() const
+    {
+        return "Joint Edit Mode";
```
[the remark this answers is not in the document]
Comment (amzn-ahmadkrm) [reply Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentMode.cpp:324]: Edit Mode seems to only modify 1 joint at a time, it's just that the other joints use the same JointsComponentMode class.

**Review by hultonha — APPROVED** [review]
LGTM! Thanks for updating these @amzn-ahmadkrm :+1:

**Comment by hultonha** [comment]
Just kicked off a build

File: Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentMode.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (hultonha) [reply Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentMode.cpp:324]: Agreed I believe this should just be 'Joint' Edit Mode

**Pull request merged**